src/content/get-started/async.md

```diff
@@ -0,0 +1,374 @@
+
+    ```dart
+    // ... (beginning of runApp function, after determining articleTitle)
+
+    void runApp(List<String>? arguments) async {
+      late String? articleTitle;
+      if (arguments == null || arguments.isEmpty) {
+        print('Please provide an article title.');
+        final inputFromStdin = stdin.readLineSync();
+        if (inputFromStdin == null || inputFromStdin.isEmpty) {
+          print('No article title provided. Exiting.');
+          return;
+        }
+        articleTitle = inputFromStdin;
+      } else {
+        articleTitle = arguments.join(' ');
+      }
+
+      print('Looking up articles about "$articleTitle". Please wait.');
+
+      // Call the API and await the result
+      var articleContent = await getWikipediaArticle(articleTitle);
+      print(articleContent); // Print the full article response (raw JSON for now)
+    }
+    ```
+    * `await getWikipediaArticle(articleTitle)`: Because `getWikipediaArticle` is an `async` function, we need to `await` its result. This pauses the `runApp` function until the `Future<String>` returned by `getWikipediaArticle` resolves into a `String` containing the article's contents.
+    * `print(articleContent)`: Prints the fetched article summary (as a raw JSON string) to the console.
+
+### Task 5: Update `main` to call `runApp`
+
+Finally, you need to update your `main` function to call the new `runApp` function when the `wikipedia` command is used.
+
+1.  Locate the `else if` block in your `main` function that currently handles the `search` command. Change the command name from `search` to `wikipedia` and update the function call.
+    In the sample code, `main` does *not* `await` the call to `runApp`, meaning `main` itself does not need to be marked `async`.
+    Your `main` function should now look like this:
+
+    ```dart
+    // ... (existing const version declaration and printUsage function)
+
+    void main(List<String> arguments) {
+      if (arguments.isEmpty || arguments.first == 'help') {
+        printUsage();
+      } else if (arguments.first == 'version') {
+        print('Dartpedia CLI version $version');
+      } else if (arguments.first == 'wikipedia') { // Changed to 'wikipedia'
+        // Pass all arguments *after* 'wikipedia' to runApp
+        final inputArgs = arguments.length > 1 ? arguments.sublist(1) : null;
+        runApp(inputArgs); // Call runApp (no 'await' needed here for main)
+      } else {
+        printUsage(); // Catch all for any unrecognized command.
+      }
+    }
+    ```
+    * `arguments.sublist(1)`: This extracts all elements from the `arguments` list starting from the second element (index 1). This effectively removes the `wikipedia` command itself, so `runApp` only receives the actual article title arguments.
+    * `runApp(inputArgs)`: We call `runApp` directly. Since `main` doesn't need to do anything after `runApp` completes, we don't need to `await` it from `main` (and therefore `main` doesn't need to be `async`).
```

Comment:
Avoid using 'we'. I'd do a search and replace.

Reply:
Fixed throughout.